core/src/main/java/io/grpc/internal/AbstractClientStream2.java

```diff
@@ -308,4 +333,69 @@ private void closeListener(Status status, Metadata trailers) {
       }
     }
   }
+
+  public static final String GRPC_PAYLOAD_BIN_KEY = "grpc-payload-bin";
+
+  private class GetFramer implements Framer {
+    private Metadata headers;
+    private boolean closed;
+    private final StatsTraceContext statsTraceCtx;
+    private byte[] payload;
+
+    public GetFramer(Metadata headers, StatsTraceContext statsTraceCtx) {
+      this.headers = Preconditions.checkNotNull(headers, "headers");
+      this.statsTraceCtx = Preconditions.checkNotNull(statsTraceCtx, "statsTraceCtx");
+    }
+
+    @Override
+    public void writePayload(InputStream message) {
+      Preconditions.checkState(payload == null, "writePayload should not be called multiple times");
+      try {
+        payload = IoUtils.toByteArray(message);
+      } catch (java.io.IOException ex) {
+        throw new RuntimeException(ex);
+      }
+    }
+
+    @Override
+    public void flush() {}
+
+    @Override
+    public boolean isClosed() {
+      return closed;
+    }
+
+    /** Closes, with flush. */
+    @Override
+    public void close() {
+      closed = true;
+      Preconditions.checkState(payload != null,
+          "Lack of request message. GET request is only supported for unary requests");
+      abstractClientStreamSink().writeHeaders(headers, payload);
+      statsTraceCtx.wireBytesSent(payload.length);
+      payload = null;
+      headers = null;
+    }
+
+    /** Closes, without flush. */
+    @Override
+    public void dispose() {
+      closed = true;
+      payload = null;
+      headers = null;
+    }
+
+    @Override
+    public Framer setMessageCompression(boolean enable) {
+      return this;
```

Comment:
Why is this not implemented? Will it ever be implemented? In any event, I'd suggest comment (TODO or explanation why it's unimplemented)